### PR TITLE
Collect transmission cfg instances

### PR DESCRIPTION
**Pull request by jontje**
As per title

### CODE REVIEW

File: src/rws_interface.cpp [hunk 1]
```diff
@@ -82,6 +82,38 @@ RWSInterface::StaticInfo RWSInterface::collectStaticInfo()
...
+    cfg::moc::Transmission transmission;
+
+    transmission.name = xmlNodeGetAttributeValue(instances[i], Identifiers::TITLE);
+
+    for (size_t j = 0; j < attributes.size(); ++j)
+    {
+      Poco::XML::Node* attribute = attributes[j];
+      if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "rotating_move"))
+      {
+        transmission.rotating_move = (xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE) == "true");
```
Comment (gavanderhoorn) [inline src/rws_interface.cpp:107]: I haven't looked up what `xmlNodeHasAttribute(..)` does exactly, but is `attribute` equal to something which encodes for revolute joints here?

**Review by gavanderhoorn — APPROVED** [review]
Looks OK.

**Comment by gavanderhoorn** [comment]
Approved assuming we'll get error handling similar to #113 here.

File: src/rws_interface.cpp (hunk 1 above)
Comment (jontje) [reply src/rws_interface.cpp]: Example of a transmission list item (collected from a virtual controller):

Capture2

The `xmlNodeHasAttribute(attribute, Identifiers::TITLE, "rotating_move")` just checks if the XML node (i.e. `attribute`) has the XML attribute `title="rotating_move"`, and `rotating_move` represents an attribute for a RobotWare transmission configuration instance.

Side note, the [getNodeByPath(...)](https://pocoproject.org/docs/Poco.XML.Node.html#32546) function can probably replace several of the custom functions for finding and checking XML nodes.

**Comment by jontje** [comment]
@gavanderhoorn can you reapprove the PR? I have synced it with the base branch, and added error handling.

**Review by gavanderhoorn — APPROVED** [review]

**Comment by jontje** [comment]
Thanks for the review @gavanderhoorn 👍

**Pull request merged**